### PR TITLE
split service.fs

### PR DESCRIPTION
**Pull request by dsyme**
This is a spike of a cleanup to split service.fs into

* `FSharpCheckerResults.fs/fsi`
* `FSharpChecker.fs/fsi`

along with a little rejigging to make the abstraction boundary cleaner (in particular TypeCheckInfo is completely internal to the former)

There's about 2000 LOC in the former and 1200 LOC in the latter.  

This work would conflict with changes in dev16.0 and I think should probably just go into that branch (I guess I'd do the work to do the merge, or redo the work completely)

### CODE REVIEW

**Comment by dsyme** [comment]
@cartermp Same test failed:

>    at FSharp.Compiler.Service.Tests.PerfTests.Test request for parse and check doesn't check whole project() in D:\a\1\s\tests\service\PerfTests.fs:line 75

**Comment by dsyme** [comment]
Two remaining test failures, I can't tell if it's transient - both TypeProvider disposal smoke tests failed.  I'll re-run tests.

**Comment by dsyme** [comment]
After discussing with @TIHan and @cartermp we will wait until dev16.0 integrates into master before merging this

**Comment by dsyme** [comment]
unrelated test failure:
```
Failed   TypeProvider.Disposal.SmokeTest1
Error Message:
   Check4b, countCreations() = countDisposals(), iteration 25
  Expected: True
  But was:  False

Stack Trace:
   at Tests.LanguageService.Script.UsingMSBuild.TypeProviderDisposalSmokeTest(Boolean clearing) in D:\a\1\s\vsintegration\tests\UnitTests\LegacyLanguageService\Tests.LanguageService.Script.fs:line 1647
   at Tests.LanguageService.Script.UsingMSBuild.TypeProvider.Disposal.SmokeTest1() in D:\a\1\s\vsintegration\tests\UnitTests\LegacyLanguageService\Tests.LanguageService.Script.fs:line 1664
```

**Comment by dsyme** [comment]
Odd unit test failure, unrelated:
```
Failed   Automation.UnionAndStructFromFSProj.UnionPropertyInComputationExpression
Error Message:
 System.IO.IOException : The process cannot access the file 'D:\a\1\s\artifacts\tmp\Release\9c8917a0-5b3b-459d-b367-94d88cbed4a0\testFile.fs' because it is being used by another process.
Stack Trace:
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at System.IO.File.Delete(String path)
   at <StartupCode$VisualFSharp-UnitTests>.$ProjectOptionsBuilder.System-IDisposable-Dispose@32.Invoke(Tuple`2 tupledArg) in D:\a\1\s\vsintegration\tests\UnitTests\ProjectOptionsBuilder.fs:line 32
   at Microsoft.FSharp.Primitives.Basics.List.iter[T](FSharpFunc`2 f, FSharpList`1 x) in D:\a\1\s\src\fsharp\FSharp.Core\local.fs:line 91
   at VisualFSharp.UnitTests.Roslyn.FSharpProject.System-IDisposable-Dispose() in D:\a\1\s\vsintegration\tests\UnitTests\ProjectOptionsBuilder.fs:line 32
   at Microsoft.VisualStudio.FSharp.Editor.Tests.Roslyn.QuickInfo.GetQuickInfoTextFromCode(String code) in D:\a\1\s\vsintegration\tests\UnitTests\QuickInfoTests.fs:line 45
   at Microsoft.VisualStudio.FSharp.Editor.Tests.Roslyn.QuickInfo.Automation.UnionAndStructFromFSProj.UnionPropertyInComputationExpression() in D:\a\1\s\vsintegration\tests\UnitTests\QuickInfoTests.fs:line 442
```

**Comment by cartermp** [comment]
@dsyme is this still WIP?

**Comment by cartermp** [comment]
Also it seems like a good opportunity to normalize more whitespace here (e.g., line breaks)

**Comment by dsyme** [comment]
> Also it seems like a good opportunity to normalize more whitespace here (e.g., line breaks)


I'll do in a separate PR - this PR is splitting a large file in half which is a bit annoying as git loses the source history tracking with the half of the file moved into the new file.  

In case we ever needed it I created couple of branches that first duplicate "service.fs" and "service.fsi" into "FSharpCheckerResults.fs" and "FSharpCheckerResults.fsi" and then applies this diff, https://github.com/dsyme/visualfsharp/compare/tmp2...clean5b

That's a more realistic view of the changes in this PR

**Comment by dsyme** [comment]
@TIHan This is ready. Could you review? 

This may be better for review: https://github.com/dsyme/visualfsharp/compare/tmp2...clean5b

**Comment by dsyme** [comment]
Unexpected unrelated failure in `Match01.fs` baseline test, runnng #6629 to check if this is a failure in master

**Comment by dsyme** [comment]
Looks like the failure from "visualfsharp-CI — #20190424.3 failed" is fake.

**Comment by dsyme** [comment]
@TIHan Can you approve?  I'd like to get this in (it is a PITA to maintain :) )

**Comment by dsyme** [comment]
@TIHan I'd like to keep working on pulling apart some of the monolithic files, I think it will help

File: src/buildfromsource/FSharp.Compiler.Private/FSharp.Compiler.Private.fsproj [hunk 1]
```diff
@@ -0,0 +1,662 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (TIHan) [inline src/buildfromsource/FSharp.Compiler.Private/FSharp.Compiler.Private.fsproj]: This shouldn't be here as `buildfromsource` is gone from master.

**Pull request merged**